cmd/flow-dps-live/main.go

```diff
@@ -171,9 +171,14 @@ func run() int {
 	// fill up fast enough. This avoids having latency between when we add data
 	// to the transaction and when it becomes available on-disk for serving the
 	// DPS API.
-	write := index.NewWriter(indexDB, storage,
-		index.WithFlushInterval(flagFlushInterval),
-	)
+	metricsEnabled := flagMetrics != ""
+	var write dps.Writer
+	if metricsEnabled {
+		write = index.NewMetricsWriter(index.NewWriter(indexDB, storage, index.WithFlushInterval(flagFlushInterval)))
+	} else {
+		write = index.NewWriter(indexDB, storage, index.WithFlushInterval(flagFlushInterval))
+	}
```

Comment:
You can just initialize the normal writer, and wrap the metrics writer conditionally:

```
var write dps.Writer
write = index.NewWriter(...)
if flagMetrics != "" {
  write = index.NewMetricsWriter(write)
}
```

Reply:
That would only work if the NewMetricsWriter took a `dps.Writer` instead of a `Writer` directly unfortunately, that's why I did not write it like this initially.

I'll change the way it is initialized though and see if I can do something more elegant.